src/GenAPI/Microsoft.DotNet.GenAPI.Shared/GenAPIApp.cs

```diff
@@ -0,0 +1,178 @@
+
+            loader.AddReferenceSearchPaths(SplitPaths(context.LibPath));
+            AddReferenceToRuntimeLibraries(loader);
+
+            var intersectionFilter = new IntersectionFilter()
+                .Add<FilterOutDelegateMembers>()
+                .Add<FilterOutImplicitSymbols>()
+                .Add(new SymbolAccessibilityBasedFilter(context.IncludeVisibleOutsideOfAssembly));
+
+            if (context.ExcludeAttributesList != null)
+            {
+                intersectionFilter.Add(new FilterOutAttributes(context.ExcludeAttributesList));
+            }
+
+            var assemblySymbols = loader.LoadAssemblies(SplitPaths(context.Assembly));
+            foreach (var assemblySymbol in assemblySymbols)
+            {
+                if (assemblySymbol == null) continue;
+
+                using var fileBuilder = new CSharpFileBuilder(
+                    intersectionFilter,
+                    GetTextWriter(context.OutputPath, assemblySymbol.Name),
+                    new CSharpSyntaxWriter(context.ExceptionMessage));
+
+                fileBuilder.WriteAssembly(assemblySymbol);
+            }
+        }
+
+        /// <summary>
+        /// Creates a TextWriter capable to write into Console or cs file.
+        /// </summary>
+        /// <param name="outputDirPath">Path to a directory where file with `assemblyName`.cs filename needs to be created.
+        ///     If Null - output to Console.Out.</param>
+        /// <param name="assemblyName">Name of an assembly. if outputDirPath is not a Null - represents a file name.</param>
+        /// <returns></returns>
+        private static TextWriter GetTextWriter(string? outputDirPath, string assemblyName)
+        {
+            if (outputDirPath == null)
+            {
+                return Console.Out;
+            }
+
+            string fileName = assemblyName + ".cs";
+            if (Directory.Exists(outputDirPath) && !string.IsNullOrEmpty(fileName))
+            {
+                return File.CreateText(Path.Combine(outputDirPath, fileName));
+            }
+
+            return File.CreateText(outputDirPath);
+        }
+
+        /// <summary>
+        /// Splits delimiter separated list of pathes represented as a string to a List of paths.
+        /// </summary>
+        /// <param name="pathSet">Delimiter separated list of paths.</param>
+        /// <returns></returns>
+        private static string[] SplitPaths(string? pathSet)
+        {
+            if (pathSet == null) return Array.Empty<string>();
+
+            return pathSet.Split(new char[] { ',', ';' }, StringSplitOptions.RemoveEmptyEntries);
+        }
+
+        /// <summary>
+        /// Read the header file if specified, or use default one.
+        /// </summary>
+        /// <param name="headerFile">File with an alternate header content to prepend to output</param>
+        /// <returns></returns>
+        public static string ReadHeaderFile(string? headerFile)
```

Comment:
Although the method isn't used today, I would assume that it should be private.

```suggestion
        private static string ReadHeaderFile(string? headerFile)
```